### DISCUSSION TITLE
API support for metrics aggregations and alarms

### DISCUSSION THREAD

**Discussion by elminsterjimmy** [apache/skywalking#8483] [Q&A] [answered]
I'm new to skywalking, and have a quick glance of the document of skywalking, I noticed that skywalking provides the functionilities of metrics aggregation (known as OAL script) and alarm (known as alram rule) based on the setting file, which are great to have, but might a liitle static for me.
I'd like to have the capacity of create / modify / list metrics aggregations and create / modify / list arlams dynamacilly by API or something similar, that I could create them within my service CD pipeline automatically.
Looking forward to having some soultion on this, and thanks in advance.

**Comment by wu-sheng** [comment, marked as answer]
Alarm rules could be set dynamically. https://skywalking.apache.org/docs/main/v8.9.1/en/setup/backend/backend-alarm/#update-the-settings-dynamically

Meanwhile, it is unrealistic to change metrics aggregation dynamically. It is a distributed aggregation, it is not good to change in the runtime or from time to time.

**Reply by elminsterjimmy** [reply]
Thanks @wu-sheng . It saves my day.
